### ISSUE TITLE
Clean up contract call gas accounting code

### ISSUE THREAD

**Issue by Ekleog-NEAR** [near/nearcore#8199]
The code around https://github.com/near/nearcore/pull/8179 is not the most obvious one. It might benefit from a complete overhaul.

**Comment by nagisa** [comment]
Is this not a duplicate of https://github.com/near/nearcore/issues/5148?

**Comment by Ekleog-NEAR** [comment]
Good point! I was thinking more along the lines of "contract call gas accounting code is hard to read, missing comments etc. here", but I’ve added a link to #5148 in the tracking issue :)
